protocol/src/lib/create-loyalty-program.ts

```diff
@@ -0,0 +1,93 @@
+import { generateSigner, KeypairSigner, PublicKey } from '@metaplex-foundation/umi'
+import { createCollection, CreateCollectionArgsPlugin, PluginAuthority } from '@metaplex-foundation/mpl-core'
+import { ATTRIBUTE_KEYS, PLUGIN_TYPES } from '@/lib/constants'
+import { VerxioContext } from '@/types/verxio-context'
+import { toBase58 } from '@/utils/to-base58'
+import { LoyaltyProgramTier } from '@/types/loyalty-program-tier'
+import { assertValidContext } from '@/utils/assert-valid-context'
+
+export interface CreateLoyaltyProgramConfig {
+  collectionSigner?: KeypairSigner
+  metadataUri: string
+  organizationName: string
+  pointsPerAction: Record<string, number>
+  programAuthority: PublicKey
+  tiers: LoyaltyProgramTier[]
+}
+
+export async function createLoyaltyProgram(
+  context: VerxioContext,
+  config: CreateLoyaltyProgramConfig,
+): Promise<{ collection: KeypairSigner; signature: string }> {
+  assertValidContext(context)
+  assertValidCreateLoyaltyProgramConfig(config)
+  const collection = config.collectionSigner ?? generateSigner(context.umi)
+
+  try {
+    const tx = await createCollection(context.umi, {
+      collection,
+      name: config.organizationName,
+      plugins: createLoyaltyProgramPlugins(config),
+      uri: config.metadataUri,
+    }).sendAndConfirm(context.umi, { confirm: { commitment: 'confirmed' } })
+
+    return { collection, signature: toBase58(tx.signature) }
+  } catch (error) {
+    throw new Error(`Failed to create loyalty program: ${error}`)
+  }
+}
+
+export function createLoyaltyProgramPlugins(config: CreateLoyaltyProgramConfig): CreateCollectionArgsPlugin[] {
+  return [
+    {
+      type: PLUGIN_TYPES.ATTRIBUTES,
+      attributeList: [
+        { key: ATTRIBUTE_KEYS.PROGRAM_TYPE, value: 'loyalty' },
+        { key: ATTRIBUTE_KEYS.TIERS, value: JSON.stringify(config.tiers) },
+        { key: ATTRIBUTE_KEYS.POINTS_PER_ACTION, value: JSON.stringify(config.pointsPerAction) },
+        { key: ATTRIBUTE_KEYS.CREATOR, value: config.programAuthority.toString() },
+      ],
+    },
+    {
+      type: PLUGIN_TYPES.PERMANENT_TRANSFER_DELEGATE,
+      authority: {
+        address: config.programAuthority,
+        type: 'Address',
+      } as PluginAuthority,
+    },
+  ]
+}
+
+// TODO: Replace with zod validation
+function assertValidCreateLoyaltyProgramConfig(
```

Comment:
As the todo says, we want to use a proper library for doing validation. Then probably pull it out of this file into its own helper, and make a separate test that only worries about validating `CreateLoyaltyProgramConfig`